visualvm/launcher/visualvm

```diff
@@ -61,9 +61,26 @@ case "`uname`" in
         ;;
 esac
 
+set_jdk_java_home_from_env_vars() {
+  # Check if JDK_HOME is set and valid
+  if [ -n "${JDK_HOME}" ] && [ -d "${JDK_HOME}" ] && [ -f "${JDK_HOME}/bin/java" ]; then
+  	visualvm_jdkhome="${JDK_HOME}"
+  # Check if JAVA_HOME is set and valid
+  elif [ -n "${JAVA_HOME}" ] && [ -d "${JAVA_HOME}" ] && [ -f "${JAVA_HOME}/bin/java" ]; then
+  	visualvm_jdkhome="${JAVA_HOME}"
+  fi
+}
+
+set_jdk_java_home_from_conf(){
```

Comment:
nit: set_jdk_java_home would be a better name, as this falls back to environment values.